devbuild.sh

```diff
@@ -1,76 +1,235 @@
 #!/bin/bash
-set -eu
-
-#cd to location of script
-MYDIR=$(cd "$(dirname "$(readlink -f -n "${BASH_SOURCE[0]}" )" )" && pwd -P)
 
+# usage instructions
 usage () {
-  echo "Usage: "
-  echo "  $0 PLATFORM COMPILER"
-  echo ""
-  echo "PLATFORM: Name of machine you are building on"
-  echo "COMPILER: (optional) compiler to use; valid options are 'intel', 'gnu'"
-  echo ""
-  echo "NOTE: This script is for internal developer use only;"
-  echo "See User's Guide for detailed build instructions"
+  printf "Usage: $0 [OPTIONS]...\n"
+  printf "\n"
+  printf "OPTIONS\n"
+  printf "  -h, --help\n"
+  printf "      show this help guide\n"
+  printf "  --platform=PLATFORM\n"
+  printf "      name of machine you are building on; defauts to hostname\n"
+  printf "      (.e.g. cheyenne | hera | jet | orion | wcoss)\n"
```

Comment:
Why the leading dot in `.e.g.`?

Reply:
I'm listing example platforms. I don't know if anyone plans to maintain the help document and keep it sync'd up with all the supported platforms. Do you want me to remove "e.g. " here? Everywhere?

Reply:
Oh sorry, the examples are fine.  I was just talking about the "." before the "e" in ".e.g.".  Should it just be "e.g."?  Don't remove the listed examples, they're very useful!

Reply:
That's funny that I didn't even notice the leading ".". I will remove it.

Reply:
There's 3 more of those below!

Reply:
Removed